### ISSUE TITLE
open-telemetry/exporter-otlp explicitly requires open-telemetry/gen-otlp-protobuf while you could be using GRPC

### ISSUE THREAD

**Issue by dkarlovi** [open-telemetry/opentelemetry-php#1342]
**Steps to reproduce**

Install `open-telemetry/exporter-otlp` and `ext-grpc`.

**What is the expected behavior?**

OTLP exporter works over GRPC so protobuf not required.

**What is the actual behavior?**

`open-telemetry/gen-otlp-protobuf` and `google/protobuf` get installed (the latter being 3.x while latest is 4.x), which are unused.

**Additional context**

This can be solved by [suggesting](https://getcomposer.org/doc/04-schema.md#suggest) the protobuf package and updating docs to install it with the exporter if it's to be the default. The current situation requires doing

```json
{
    "provide": {
        "open-telemetry/gen-otlp-protobuf": "^1.0"
    },
}
```

in your app's Composer config which works, but kind of finicky.

**Comment by brettmc** [comment]
Hi @dkarlovi 
I might be missing something, but protobuf is the only supported encoding method for OTLP over gRPC: https://opentelemetry.io/docs/specs/otlp/#protocol-details

> This specification defines how OTLP is implemented over [gRPC](https://grpc.io/) and HTTP 1.1 transports and specifies [Protocol Buffers schema](https://developers.google.com/protocol-buffers/docs/overview) that is used for the payloads.

`open-telemetry/gen-otlp-protobuf` contains the opentelemetry generated protobuf schema described above, for communicating with an OTLP-compatible backend.

**Comment by dkarlovi** [comment]
Hey @brettmc, you're right, I've misunderstood the layout of the packages. In the end the offending part was actually `google/protobuf`, the latest version is 4 and the package allows ^3 only.

I've fixed this by updating my composer.json to:

```json
    "provide": {
        "google/protobuf": "^3.0"
    },
```

This removes the fallback which is not required since I have `ext-protobuf` installed.

I'm not sure what would be the best solution for the library itself, but this works for me. Closing.
